File: 0692-top-k-frequent-words/0692-top-k-frequent-words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map<string, int> count;
        
        for (string& word : words) {
            count[word]++;
        }
        
        auto cmp = [](pair<string,int>& a, pair<string,int>& b) {
            if (a.second != b.second) {
                return a.second > b.second;
            }
            return a.first < b.first;
        };
        
        vector<pair<string,int>> freqList(count.begin(), count.end());
        sort(freqList.begin(), freqList.end(), cmp);
        
        vector<string> result;
        for (int i = 0; i < k; i++) {
            result.push_back(freqList[i].first);
        }
        
        return result;
    }
};

File: 0692-top-k-frequent-words/0692-top-k-frequent-words.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        Map<String, Integer> count = new HashMap<>();
        
        for (String word : words) {
            count.put(word, count.getOrDefault(word, 0) + 1);
        }
        
        PriorityQueue<String> pq = new PriorityQueue<>((a, b) -> {
            if (!count.get(a).equals(count.get(b))) {
                return count.get(a) - count.get(b);
            }
            return b.compareTo(a);
        });
        
        for (String word : count.keySet()) {
            pq.offer(word);
            
            if (pq.size() > k) {
                pq.poll();
            }
        }
        
        List<String> result = new ArrayList<>();
        while (!pq.isEmpty()) {
            result.add(pq.poll());
        }
        
        Collections.reverse(result);
        return result;
    }
}